packages/react-query/src/types.ts

```diff
@@ -36,6 +37,16 @@ export interface UseQueryOptions<
     'queryKey'
   > {}
 
+export interface UseSuspenseQueryOptions<
+  TQueryFnData = unknown,
+  TError = DefaultError,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+> extends Omit<
+    UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
+    'enabled' | 'suspense' | 'throwOnError' | 'placeholderData'
+  > {}
```

Comment:
I thought It would be better that useSuspenseQuery have its `QueryOptions` type and `Result` type like `UseQueryOptions`, `UseQueryResult` for useQuery

If useSuspenseQuery should be provided as an API of the same hierarchy as useQuery, I thought it would be better to provide the types(`UseSuspenseQueryOptions`, `UseSuspenseQueryResult`) in the same hierarchy.